### PR TITLE
fixes #6400 - show * next to required fields on forms

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by isratrade** [comment]
Name \* [text box] with a space per kyle's suggestion.  This commit was previously part of PR #1512

File: app/helpers/layout_helper.rb [hunk 1]
```diff
@@ -150,10 +150,11 @@ def field(f, attr, options = {})
       content_tag :div, :class => "form-group #{error.empty? ? "" : 'has-error'}",
                   :id          => options.delete(:control_group_id) do
 
+        required_mark = ' *'.html_safe if is_required?(f, attr)
```
Comment (isratrade) [inline app/helpers/layout_helper.rb]: the space is added here

**Comment by dLobatog** [comment]
It works well and :+1: to the code, however there are some oddities because of the .html_safe, for instance in ConfigTemplates#edit it displays labels.

File: app/helpers/layout_helper.rb [hunk 2]
```diff
@@ -163,6 +164,11 @@ def field(f, attr, options = {})
     end
   end
 
+  def is_required?(f, attr)
```
Comment (domcleal) [inline app/helpers/layout_helper.rb:167]: I think this method deserves a unit test

**Comment by dLobatog** [comment]
I think the redmine issue is linked to the wrong PR too.

**Comment by isratrade** [comment]
@eLobato, re-committed with fix for label "Applicable Operating Systems" and added test for `is_required?(f, attr)`

File: app/helpers/layout_helper.rb (hunk 2 above)
Comment (dLobatog) [inline app/helpers/layout_helper.rb:167]: [fixed]

**Comment by dLobatog** [comment]
Thanks, I checked forms for all models and I found no other hiccups. 
However in models with validators that check presence without `:presence => true`, this won't work. For instance, try creating a Filter without any permissions. Permissions isn't showed with an asterisk but an empty value won't work. 
I'm okay merging this as is, as finding whether an attribute presence or not is mandatory without the aid of checking if it responds to validates_on might be too much work to just show an asterisk. Comments?

**Comment by domcleal** [comment]
I'd suggest adding something to the options hash on the filter form (e.g. `:required => true`) and checking it in the field helper, overriding `is_required` if it's set.

**Comment by isratrade** [comment]
@eLobato, @domcleal, added new commit to be squashed 
Added option on form helpers for `:required => true`

**Comment by dLobatog** [comment]
That looks like a good option, :+1: for me, I'll test it tomorrow to see if we're missing `:required => true` in any other forms.

**Comment by isratrade** [comment]
@eLobato, I squash the 2nd commit after you reviewed it, so it should be ready for merge. I search for more has_many relationships that are required, but didn't find any.

**Comment by dLobatog** [comment]
Looks clean and I couldn't find any other form with errors. Merging in 3.. 2.. 1

**Comment by dLobatog** [comment]
Merged as https://github.com/theforeman/foreman/commit/2af7c64a3b9c2699a3131483bc2344b50c138542 , thanks @isratrade !

**Pull request closed without merging**